openapi3 test fails in CI due to failed host resolution

In `openapi3` package, the `TestExtraSiblingsInRemoteRef` test now fails locally and in CI because one of the endpoints we depend on for testing has started to fail to resolve.

Error msg:
```
--- FAIL: TestExtraSiblingsInRemoteRef (0.00s)
    --- FAIL: TestExtraSiblingsInRemoteRef/'3.0' (0.01s)
        issue513_test.go:73:
            	Error Trace:	/Users/alexawang/Repos/kin-openapi/openapi3/issue513_test.go:73
            	Error:      	Received unexpected error:
            	            	Get "http://schemas.sentex.io/store/categories.json": dial tcp: lookup schemas.sentex.io: no such host
            	Test:       	TestExtraSiblingsInRemoteRef/'3.0'
    --- FAIL: TestExtraSiblingsInRemoteRef/'3.1' (0.01s)
        issue513_test.go:73:
            	Error Trace:	/Users/alexawang/Repos/kin-openapi/openapi3/issue513_test.go:73
            	Error:      	Received unexpected error:
            	            	Get "http://schemas.sentex.io/store/categories.json": dial tcp: lookup schemas.sentex.io: no such host
            	Test:       	TestExtraSiblingsInRemoteRef/'3.1'
FAIL
```

We may want to remove this remote, or find another remote that fulfills this test.

I checked the remote resolution locally with curl as well which, as expected, resulted in:
```
* Could not resolve host: schemas.sentex.io
* Closing connection
curl: (6) Could not resolve host: schemas.sentex.io
```

--- a/openapi3/issue513_test.go
+++ b/openapi3/issue513_test.go
@@ -2,6 +2,7 @@ package openapi3
 
 import (
 	"encoding/json"
+	"net/url"
 	"strings"
 	"testing"
 
@@ -34,40 +35,47 @@ paths:
                 $ref: http://schemas.sentex.io/store/categories.json
 `
 
-	// When that site fails to respond:
-	// see https://github.com/getkin/kin-openapi/issues/495
+	resolver := func(loader *Loader, url *url.URL) (data []byte, err error) {
+		switch url.String() {
+		case "http://schemas.sentex.io/store/categories.json":
+			data = []byte(`
+			{
+			  "$id": "http://schemas.sentex.io/store/categories.json",
+			  "$schema": "http://json-schema.org/draft-07/schema#",
+			  "description": "array of category strings",
+			  "type": "array",
+			  "items": {
+			    "allOf": [
+			      {
+			        "$ref": "http://schemas.sentex.io/store/category.json"
+			      }
+			    ]
+			  }
+			}`)
 
-	// http://schemas.sentex.io/store/categories.json
-	// {
-	//   "$id": "http://schemas.sentex.io/store/categories.json",
-	//   "$schema": "http://json-schema.org/draft-07/schema#",
-	//   "description": "array of category strings",
-	//   "type": "array",
-	//   "items": {
-	//     "allOf": [
-	//       {
-	//         "$ref": "http://schemas.sentex.io/store/category.json"
-	//       }
-	//     ]
-	//   }
-	// }
+		case "http://schemas.sentex.io/store/category.json":
+			data = []byte(`
+			{
+			  "$id": "http://schemas.sentex.io/store/category.json",
+			  "$schema": "http://json-schema.org/draft-07/schema#",
+			  "description": "category name for products",
+			  "type": "string",
+			  "pattern": "^[A-Za-z0-9\\-]+$",
+			  "minimum": 1,
+			  "maximum": 30
+			}`)
 
-	// http://schemas.sentex.io/store/category.json
-	// {
-	//   "$id": "http://schemas.sentex.io/store/category.json",
-	//   "$schema": "http://json-schema.org/draft-07/schema#",
-	//   "description": "category name for products",
-	//   "type": "string",
-	//   "pattern": "^[A-Za-z0-9\\-]+$",
-	//   "minimum": 1,
-	//   "maximum": 30
-	// }
+		default:
+			panic(url)
+		}
+		return
+	}
 
-	for _, majmin := range []string{"'3.0'", "'3.1'"} {
+	for _, majmin := range []string{"'3.0'", "'3.1'", "'3.2'"} {
 		t.Run(majmin, func(t *testing.T) {
 			t.Parallel()
 			sl := NewLoader()
-			sl.IsExternalRefsAllowed = true
+			sl.ReadFromURIFunc = resolver
 
 			doc, err := sl.LoadFromData([]byte(strings.ReplaceAll(spec, "3.0.1", majmin)))
 			require.NoError(t, err)

--- a/openapi3/issue513_test.go
+++ b/openapi3/issue513_test.go
@@ -1,4 +1,4 @@
-package openapi3
+package openapi3_test
 
 import (
 	"encoding/json"
@@ -7,6 +7,8 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/require"
+
+	"github.com/getkin/kin-openapi/openapi3"
 )
 
 func TestExtraSiblingsInRemoteRef(t *testing.T) {
@@ -35,7 +37,7 @@ paths:
                 $ref: http://schemas.sentex.io/store/categories.json
 `
 
-	resolver := func(loader *Loader, url *url.URL) (data []byte, err error) {
+	resolver := func(loader *openapi3.Loader, url *url.URL) (data []byte, err error) {
 		switch url.String() {
 		case "http://schemas.sentex.io/store/categories.json":
 			data = []byte(`
@@ -74,13 +76,13 @@ paths:
 	for _, majmin := range []string{"'3.0'", "'3.1'", "'3.2'"} {
 		t.Run(majmin, func(t *testing.T) {
 			t.Parallel()
-			sl := NewLoader()
+			sl := openapi3.NewLoader()
 			sl.ReadFromURIFunc = resolver
 
 			doc, err := sl.LoadFromData([]byte(strings.ReplaceAll(spec, "3.0.1", majmin)))
 			require.NoError(t, err)
 
-			err = doc.Validate(sl.Context, AllowExtraSiblingFields("$id", "$schema"))
+			err = doc.Validate(sl.Context, openapi3.AllowExtraSiblingFields("$id", "$schema"))
 			require.NoError(t, err)
 		})
 	}
@@ -122,7 +124,7 @@ components:
 	for _, majmin := range []string{"3.0", "3.1"} {
 		t.Run(majmin, func(t *testing.T) {
 			t.Parallel()
-			sl := NewLoader()
+			sl := openapi3.NewLoader()
 			doc, err := sl.LoadFromData([]byte(strings.ReplaceAll(spec, "3.0.3", majmin)))
 			require.NoError(t, err)
 			err = doc.Validate(sl.Context)
@@ -173,7 +175,7 @@ components:
 	for _, majmin := range []string{"3.0", "3.1"} {
 		t.Run(majmin, func(t *testing.T) {
 			t.Parallel()
-			sl := NewLoader()
+			sl := openapi3.NewLoader()
 			doc, err := sl.LoadFromData([]byte(strings.ReplaceAll(spec, "3.0.3", majmin)))
 			require.NoError(t, err)
 			require.Contains(t, doc.Paths.Value("/v1/operation").Delete.Responses.Default().Value.Extensions, `x-my-extension`)
@@ -220,7 +222,7 @@ components:
 	for _, majmin := range []string{"3.0", "3.1"} {
 		t.Run(majmin, func(t *testing.T) {
 			t.Parallel()
-			sl := NewLoader()
+			sl := openapi3.NewLoader()
 			doc, err := sl.LoadFromData([]byte(strings.ReplaceAll(spec, "3.0.3", majmin)))
 			require.NoError(t, err)
 			err = doc.Validate(sl.Context)
@@ -266,10 +268,10 @@ components:
 	for _, majmin := range []string{"3.0", "3.1"} {
 		t.Run(majmin, func(t *testing.T) {
 			t.Parallel()
-			sl := NewLoader()
+			sl := openapi3.NewLoader()
 			doc, err := sl.LoadFromData([]byte(strings.ReplaceAll(spec, "3.0.3", majmin)))
 			require.NoError(t, err)
-			err = doc.Validate(sl.Context, AllowExtraSiblingFields("description"))
+			err = doc.Validate(sl.Context, openapi3.AllowExtraSiblingFields("description"))
 			require.NoError(t, err)
 		})
 	}

